src/compiler/types.ts

```diff
@@ -3005,7 +3005,24 @@ namespace ts {
         isRestParameter?: boolean;
     }
 
-    export type SymbolTable = Map<Symbol>;
+    export type EscapedIdentifier = (string & { __escapedIdentifier: void }) | (void & { __escapedIdentifier: void });

```

Comment:
Mmmm, the name's actually a little hard, since a plain escaped string I would expect to be a string where `"` has been replaced with `\"` to escape it (which we do! elsewhere!). I called it `EscapedIdentifier` at first since Identifier was the only _documented_ place where we stored strings escaped this way - except that was by no means true. From identifiers, it leaked into symbol names (which was desirable, based on use), and from there, even string literals were (sometimes and inconsistently) escaped to be compatible with symbol names. All the 
 inconsistency is gone now, but the name definitely no longer makes sense. What about `UnderscoreEscapedString`?